Specifying CPU variant in verilator

Specifying the cpu variant in a Verilator simulation is not working:
`make PLATFORM=sim load EXTRA_LITEX_ARGS="--cpu-variant=fomu"`
Error says --cpu-variant is not a valid parameter.

On `main`, the general mechanism seems to work correctly.   I've run with these:

`make PLATFORM=sim load`

`make PLATFORM=sim load EXTRA_LITEX_ARGS="--cpu-variant=slim+cfu"`

And with #550,

`make PLATFORM=sim load EXTRA_LITEX_ARGS="--cpu-variant=minimal+cfu"`

And they seem to work correctly; the cycle count is 10x when using the minimal variant that doesn't have a multiplier.


But using `fomu+cfu` does give an error:

```
INFO:SoC:--------------------------------------------------------------------------------
INFO:SoC:Initial SoC:
INFO:SoC:--------------------------------------------------------------------------------
INFO:SoC:32-bit wishbone Bus, 4.0GiB Address Space.
INFO:SoC:32-bit CSR Bus, 32-bit Aligned, 16.0KiB Address Space, 2048B Paging, big Ordering (Up to 32 Locations).
INFO:SoC:IRQ Handler (up to 32 Locations).
INFO:SoC:--------------------------------------------------------------------------------
ERROR:SoC:fomu+cfu CPU variant not supported, supporteds: minimal, minimal+debug, minimal+debug+hwbp, lite, lite+debug, lite+debug+hwbp, standard, standard+debug, imac, imac+debug, full, full+cfu, full+debug, full+cfu+debug, linux, linux+debug, linux+no-dsp, secure, secure+debug, custom, custom+cfu, hps+cfu, hpsdelta+cfu, dbpl8+cfu, minimal+cfu, slim+cfu, slimopt+cfu, slim+cfu+debug, perf+cfu, perf+cfu+debug, slimperf+cfu, slimperf+cfu+debug.
make[1]: *** [/media/tim/GIT/google/CFU-Playground/soc/sim.mk:68: build/sim.proj_template_v/software/bios/bios.bin] Error 1
make[1]: Leaving directory '/media/tim/GIT/google/CFU-Playground/soc'
make: *** [../proj.mk:289: litex-software] Error 2
```
I suspect that the sim flow isn't calling `patch_cpu_fomu_variant()` in `soc/board_specific_workflows/ice40up5k.py`.

--- a/soc/board_specific_workflows/ice40up5k.py
+++ b/soc/board_specific_workflows/ice40up5k.py
@@ -23,30 +23,6 @@ from litex.soc.integration import builder
 from typing import Callable
 
 
-def patch_cpu_fomu_variant():
-    """Monkey patches the fomu variant into LiteX."""
-    core.CPU_VARIANTS.update({
-        'fomu': 'VexRiscv_Fomu',
-        'fomu+cfu': 'VexRiscv_FomuCfu'
-    })
-    core.GCC_FLAGS.update({
-        'fomu': '    -march=rv32im -mabi=ilp32 -mno-div',
-        'fomu+cfu': '-march=rv32im -mabi=ilp32 -mno-div',
-    })
-
-    old_add_soc_components = core.VexRiscv.add_soc_components
-
-    def new_add_soc_components(self, soc, soc_region_cls):
-        old_add_soc_components(self, soc, soc_region_cls)
-        if 'fomu' in self.variant:
-            soc.add_config('CPU_DIV_UNIMPLEMENTED')
-
-            # Fomu variant has *no* Dcache.
-            # This is here to avoid the dcache flush instruction (system.h).
-            soc.constants.pop('CONFIG_CPU_HAS_DCACHE', None)
-
-    core.VexRiscv.add_soc_components = new_add_soc_components
-
 
 class Ice40UP5KWorkflow(general.GeneralSoCWorkflow):
     """Workflow for boards derived from the Ice40UP5K.
@@ -60,9 +36,6 @@ class Ice40UP5KWorkflow(general.GeneralSoCWorkflow):
                  soc_constructor: Callable[..., litex_soc.LiteXSoC],
                  builder_constructor: Callable[..., builder.Builder] = None,
                  warn: bool = True) -> None:
-
-        # Remove if/when the new CPU variant is upstreamed.
-        patch_cpu_fomu_variant()
 
         if warn and args.cpu_variant != 'fomu+cfu':
             warnings.warn('Only fomu+cfu variant of Vexriscv supported for ' +

--- a/soc/patch_cpu_variant.py
+++ b/soc/patch_cpu_variant.py
@@ -27,6 +27,8 @@ import os
 def patch_cpu_variant():
     """Monkey patches custom variants into LiteX."""
     core.CPU_VARIANTS.update({
+        'fomu':                 'VexRiscv_Fomu',
+        'fomu+cfu':             'VexRiscv_FomuCfu',
         'custom':               'VexRiscv_Custom',
         'custom+cfu':           'VexRiscv_CustomCfu',
         'hps+cfu':              'VexRiscv_HpsCfu',
@@ -42,6 +44,8 @@ def patch_cpu_variant():
         'slimperf+cfu+debug':   'VexRiscv_SlimPerfCfuDebug',
     })
     core.GCC_FLAGS.update({
+        'fomu':                 '-march=rv32im -mabi=ilp32 -mno-div',
+        'fomu+cfu':             '-march=rv32im -mabi=ilp32 -mno-div',
         'custom':               '-march=rv32im -mabi=ilp32',
         'custom+cfu':           '-march=rv32im -mabi=ilp32',
         'hps+cfu':              '-march=rv32im -mabi=ilp32',
@@ -62,8 +66,16 @@ def patch_cpu_variant():
 
     def new_add_soc_components(self, soc, soc_region_cls):
         old_add_soc_components(self, soc, soc_region_cls)
+
         if 'perf' in self.variant:
             soc.add_config('CPU_PERF_CSRS', 8)
+
+        if 'fomu' in self.variant:
+            soc.add_config('CPU_DIV_UNIMPLEMENTED')
+
+            # Fomu variant has *no* Dcache.
+            # This is here to avoid the dcache flush instruction (system.h).
+            soc.constants.pop('CONFIG_CPU_HAS_DCACHE', None)
 
     core.VexRiscv.add_soc_components = new_add_soc_components
 

